rust/models/src/credential.rs

```diff
@@ -49,12 +49,22 @@ impl Default for Credential {
             credential_type: CredentialType::UP {
                 username: "root".to_string(),
                 password: "".to_string(),
-                privilege_credential: None,
+                privilege: None,
             },
         }
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde_support",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct PrivilegeInformation {
+    pub privilege_username: String,
```

Comment:
when you use `#[serde(rename = "privilege_username")]` you can rename the field to username and password respectively as it is clear within rust that those are privilege from the struct name.